packages/botfuel-dialog/src/dialog-manager.js

```diff
@@ -238,46 +247,56 @@ class DialogManager extends Resolver {
         data: {},
       });
     } else {
-      const action = await this.resolve(dialog.name).execute(adapter, userMessage, dialog.data);
+      const { action, botMessages: newBotMessages } = await this.resolve(dialog.name).execute(
```

Comment:
maybe introduce variable dialog = this.resolve(dialog.name)

Reply:
done